ntoskrnl/config/cmsysini.c

```diff
@@ -621,9 +621,14 @@ CmpCreateControlSet(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
     ASSERT(Disposition == REG_CREATED_NEW_KEY);
 
     /* Initialize the target link name */
-    RtlStringCbPrintfW(UnicodeBuffer, sizeof(UnicodeBuffer),
+    Status = RtlStringCbPrintfW(UnicodeBuffer, sizeof(UnicodeBuffer),
                        L"\\Registry\\Machine\\System\\ControlSet%03ld",
                        ControlSet);
+    if (!NT_SUCCESS(Status))
+    {
+        return Status;
```

Comment:
Leaking KeyHandle

Reply:
I refactored the code in my latest commit to use a single exit path (through the Cleanup label) that closes all open handles.